staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -212,7 +212,7 @@ func (c *codec) Encode(obj runtime.Object, w io.Writer) error {
 	// Perform a conversion if necessary
 	objectKind := obj.GetObjectKind()
 	old := objectKind.GroupVersionKind()
-	out, err := c.convertor.ConvertToVersion(obj, c.encodeVersion)
+	out, err := c.convertor.ConvertToVersion(obj.DeepCopyObject(), c.encodeVersion)
```

Comment:
@liggit @smarterclayton @deads2k the converter is not guaranteed to not mutate `obj`. We have conversions that modify annotations in a shallowly copied ObjectMeta. Noticed this because I added annotation fuzzing. In our roundtrip tests we actually test this that `runtime.Encode` does not mutate. But without this deepcopy, it does.

Reply:
won't this destroy performance? if the converter interface says it can modify the object, why do we need to do this?